processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -219,6 +247,36 @@ public SearchQueryRunner(Segment segment)
     return makeReturnResult(limit, retVal);
   }
 
+  private int getStartIndexOfTime(GenericColumn timeValues, long time, boolean inclusive)
+  {
+    int low = 0;
+    int high = timeValues.length() - 1;
+
+    while (low <= high) {
+      int mid = (low + high) >>> 1;
+      long midVal = timeValues.getLongSingleValueRow(mid);
+
+      if (midVal < time)
+        low = mid + 1;
+      else if (midVal > time)
+        high = mid - 1;
+      else { // key found
+        int i;
+        // rewind the index of the same time values
+        for (i = mid - 1; i >= 0; i--) {
+          long prev = timeValues.getLongSingleValueRow(i);
+          if (time != prev) {
+            break;
+          }
+        }
+        return inclusive ? i + 1 : i;
+      }
+    }
+    // key not found.
+    // return insert index
+    return low;
```

Comment:
@nishantmonu51 You're definitely right.
I updated source code and added some unit tests.
Thank you.